File: little-bee/controller/router/action_group.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_ACTION_GROUP = "/config/action-group"
const URL_ROUTER_ACTION_GROUP_BY_ID = "/config/action-group/:action-group-id"
const URL_ROUTER_ACTION_GROUP_QUERY = "/config/action-group/query"
const URL_ROUTER_ACTION_GROUP_ALL = "/config/action-group/all"

func ActionGroupRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_ACTION_GROUP, config.CreateActionGroup)
	r.GET(URL_ROUTER_ACTION_GROUP_BY_ID, config.RetrieveActionGroup)
	r.PUT(URL_ROUTER_ACTION_GROUP, config.UpdateActionGroup)
	r.DELETE(URL_ROUTER_ACTION_GROUP_BY_ID, config.DeleteActionGroup)
	r.PUT(URL_ROUTER_ACTION_GROUP_BY_ID, config.DeleteActionGroupWithFlag)
	r.POST(URL_ROUTER_ACTION_GROUP_QUERY, config.QueryActionGroup)
	r.GET(URL_ROUTER_ACTION_GROUP_ALL, config.GetAllActionGroups)
	return r
}

File: frontend/console/src/api/action-group.js
import Axios from "@/http";

export function createActionGroup(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/action-group', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveActionGroup(actionGroupId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/action-group/${ actionGroupId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateActionGroup(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/action-group', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteActionGroup(actionGroupId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/action-group/${ actionGroupId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteActionGroupWithFlag(actionGroupId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/action-group/${ actionGroupId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllActionGroups() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/action-group/all').then(res => {
            resolve(res)
        })
    })
}

export function queryActionGroup (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/action-group/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}